source/simulator/checkpoint_restart.cc

```diff
@@ -96,10 +111,127 @@ namespace aspect
     void load_and_check_critical_parameters (const Parameters<dim> &parameters,
                                              aspect::iarchive &ia)
     {
+      bool convert_to_years;
+      ia >> convert_to_years;
+      AssertThrow (convert_to_years == parameters.convert_to_years,
+                   ExcMessage ("The value provided for `Use years in output instead of seconds' that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      std::string output_directory;
+      ia >> output_directory;
+      AssertThrow (output_directory == parameters.output_directory,
+                   ExcMessage ("The name of the output directory that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      double surface_pressure;
+      ia >> surface_pressure;
+      AssertThrow (surface_pressure == parameters.surface_pressure,
+                   ExcMessage ("The value of surface pressure that was stored "
+                               "in the checkpoint file is not the same as the one "
+                               "you currently set in your input file. "
+                               "These need to be the same during restarting "
+                               "from a checkpoint."));
+
+      bool use_operator_splitting;
+      ia >> use_operator_splitting;
```

Comment:
is this something we can/should be able to change?

Reply:
@tjhei -- here, too, it seemed like a thing we're at best unsure whether it works. If someone really wants to do this, we can always reconsider this, but it seems like a dubious proposition.

Or do you feel strongly about it?